src/gluonnlp/data/glue.py

```diff
@@ -0,0 +1,585 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=line-too-long
+"""CoNLL format corpora."""
+
+__all__ = ['GlueCoLA', 'GlueSST2', 'GlueSTSB', 'GlueQQP', 'GlueRTE', 'GlueMNLI',
+           'GlueQNLI', 'GlueWNLI']
+
+import zipfile
+import os
+
+from mxnet.gluon.utils import download, check_sha1, _get_repo_file_url
+
+from .dataset import TSVDataset
+from .registry import register
+from ..base import get_home_dir
+
+_glue_s3_uri = 's3://apache-mxnet/gluon/dataset/Glue/'
+
+class _GlueDataset(TSVDataset):
+    def __init__(self, root, data_file, **kwargs):
+        root = os.path.expanduser(root)
+        if not os.path.isdir(root):
+            os.makedirs(root)
+        self._root = root
```

Comment:
nit: user may specify the root to be a path with `~` in it, so it might be a good idea to do `os.path.expanduser` here.

Reply:
Anything else needed besides `os.path.expanduser(root)` at line 39?

Reply:
false alarm as I missed line 39